media_remover/src/test/scala/MediaNotRequiredMessageProcessorSpec.scala

```diff
@@ -258,124 +498,1657 @@ class MediaNotRequiredMessageProcessorSpec extends Specification with Mockito {
+      implicit val mockS3ObjectChecker = mock[S3ObjectChecker]
+      implicit val mockChecksumChecker = mock[ChecksumChecker]
+      val mockAssetFolderLookup = mock[AssetFolderLookup]
+
+      val toTest = new MediaNotRequiredMessageProcessor(mockAssetFolderLookup)
+
+      val msgContent =
+        """{
+          |"mediaTier": "NEARLINE",
+          |"projectIds": ["22"],
+          |"originalFilePath": "/srv/Multimedia2/Media Production/Assets/Multimedia_Reactive_News_and_Sport/Reactive_News_Explainers_2022/monika_cvorak_MH_Investigation/Footage Vera Productions/2022-03-18_MH.mp4",
+          |"nearlineId": "1",
+          |"vidispineItemId": "VX-151922",
+          |"mediaCategory": "Deliverables"
+          |}""".stripMargin
+
+      val msgObj = io.circe.parser.parse(msgContent).flatMap(_.as[OnlineOutputMessage]).right.get
+
+      val result = Try {
+        Await.result(toTest.storeDeletionPending(msgObj), 2.seconds)
+      }
+
+      result must beSuccessfulTry
+      result.get must beRight(234)
+    }
+
+
+    "store updated record for NEARLINE if record already present" in {
+      val mockMsgFramework = mock[MessageProcessingFramework]
+      implicit val pendingDeletionRecordDAO :PendingDeletionRecordDAO = mock[PendingDeletionRecordDAO]
+      val existingRecord = PendingDeletionRecord(Some(234), "some/file/path", Some("nearline-test-id"), Some("vsid"), MediaTiers.NEARLINE, 1)
+      val expectedUpdatedRecordToSave = PendingDeletionRecord(Some(234), "some/file/path", Some("nearline-test-id"), Some("vsid"), MediaTiers.NEARLINE, 2)
+      pendingDeletionRecordDAO.findByNearlineIdForNEARLINE(any) returns Future(Some(existingRecord))
+      pendingDeletionRecordDAO.writeRecord(expectedUpdatedRecordToSave) returns Future(234)
+
+      implicit val vidispineCommunicator = mock[VidispineCommunicator]
+      implicit val mat:Materializer = mock[Materializer]
+      implicit val sys:ActorSystem = mock[ActorSystem]
+      implicit val mockBuilder = mock[MXSConnectionBuilderImpl]
+      implicit val mockS3ObjectChecker = mock[S3ObjectChecker]
+      implicit val mockChecksumChecker = mock[ChecksumChecker]
+      val mockAssetFolderLookup = mock[AssetFolderLookup]
+
+      val toTest = new MediaNotRequiredMessageProcessor(mockAssetFolderLookup)
+
+      val msgContent =
+        """{
+          |"mediaTier": "NEARLINE",
+          |"projectIds": ["22"],
+          |"originalFilePath": "/srv/Multimedia2/Media Production/Assets/Multimedia_Reactive_News_and_Sport/Reactive_News_Explainers_2022/monika_cvorak_MH_Investigation/Footage Vera Productions/2022-03-18_MH.mp4",
+          |"nearlineId": "1",
+          |"vidispineItemId": "VX-151922",
+          |"mediaCategory": "Deliverables"
+          |}""".stripMargin
+
+      val msgObj = io.circe.parser.parse(msgContent).flatMap(_.as[OnlineOutputMessage]).right.get
+
+      val result = Try {
+        Await.result(toTest.storeDeletionPending(msgObj), 2.seconds)
+      }
+
```

Comment:
again should probably have a

```scala
there was one(pendingDeletionRecoordDAO).writeRecord(expectedUpdatedReturnToSave)
```

I realise that it won't change the outcome of the test, but it's easier to read that way

Reply:
Yeap!